### ISSUE TITLE
GCop 306

### ISSUE THREAD

**Issue by marjanjavid** [Geeksltd/GCop#130]
This rule warning is not kicked.
![306](https://user-images.githubusercontent.com/20684649/39660233-8f43fb20-504f-11e8-822a-ff71443487ae.PNG)

**Comment by Foroughi** [comment]
@marjanjavid  I don't have access to that `IsBefore `method. Would you please confirm its an extension method in the context of given sample?

**Comment by marjanjavid** [comment]
Yes. It is an extension method but I did not write it myself. It is written in MSharpExtensions class.

**Comment by Foroughi** [comment]
@reporter : Fixed on [NuGetPackages\1b7b834c78847371abc1226811d1cdead2697198](https://gitlab.com/Paymon/GCop/tree/1b7b834c78847371abc1226811d1cdead2697198)

Note : In order to fix this i had to change the logic of this rule , so that would be great if we have a deep test on this rule again to make sure i didn't break any other logic of this rule
